eclair-core/src/test/scala/fr/acinq/eclair/message/OnionMessagesSpec.scala

```diff
@@ -0,0 +1,218 @@
+    )
+    val onionForAlice = OnionMessage(blindingSecret.publicKey, packet)
+
+    /*
+    *  Building the onion with functions from `OnionMessages`
+    */
+    val replyPath = OnionMessages.buildRoute(blindingOverride, IntermediateNode(carol.publicKey, padding = Some(hex"0000000000000000000000000000000000000000000000000000000000000000000000")) :: Nil, Left(Recipient(dave.publicKey, secret = Some(hex"01234567"))))
+    assert(replyPath == routeFromCarol)
+    val (_, message) = OnionMessages.buildMessage(sessionKey, blindingSecret, IntermediateNode(alice.publicKey) :: IntermediateNode(bob.publicKey) :: Nil, Right(replyPath), Nil)
+    assert(message == onionForAlice)
+
+    /*
+    *  Checking that the onion is relayed properly
+    */
+    OnionMessages.process(alice, onionForAlice) match {
+      case OnionMessages.RelayMessage(nextNodeId, onionForBob) =>
+        assert(nextNodeId == bob.publicKey)
+        OnionMessages.process(bob, onionForBob) match {
+          case OnionMessages.RelayMessage(nextNodeId, onionForCarol) =>
+            assert(nextNodeId == carol.publicKey)
+            OnionMessages.process(carol, onionForCarol) match {
+              case OnionMessages.RelayMessage(nextNodeId, onionForDave) =>
+                assert(nextNodeId == dave.publicKey)
+                OnionMessages.process(dave, onionForDave) match {
+                  case OnionMessages.ReceiveMessage(_, _) => ()
```

Comment:
Isn't there a `pathId` we can verify here?

Reply:
Done